tensorflow_quantum/core/qsim/fuser_basic.cc

```diff
@@ -31,17 +31,75 @@ using ::tensorflow::Status;
 // Appends to `fused_gates` all single-qubit gates on the current `qubit_wire`,
 // until reaching the end of the wire or a multi-qubit gate.
 // Starts the search at `current_timeslice`.
-void Advance(const std::vector<const Gate*>& qubit_wire,
-             std::vector<const Gate*>* fused_gates,
+void Advance(const std::vector<const Gate*>& qubit_wire, GateFused* gate,
              unsigned int* current_timeslice) {
   while (*current_timeslice < qubit_wire.size() &&
          qubit_wire[*current_timeslice]->num_qubits == 1) {
-    fused_gates->push_back(qubit_wire[(*current_timeslice)++]);
+    gate->AddGate(qubit_wire[(*current_timeslice)++]);
   }
 }
 
 }  // namespace
 
+GateFused::GateFused(const unsigned int time, const unsigned int q0,
+                     const unsigned int q1, const Gate* anchor)
+    : time_(time), anchor_(anchor) {
+  qubits_[0] = q0;
+  qubits_[1] = q1;
+}
+
+void GateFused::AddGate(const Gate* gate) { gates_.push_back(gate); }
+
+std::vector<const Gate*> GateFused::GetAllGates() const { return gates_; }
```

Comment:
Does this return the object that contains the pointers, adding some unnecessary overhead ? It might be better to do something like 
```C++
GetAllGates(std::vector<const Gate*>* ref) { ref = &gates_; }
```

Reply:
Switched to parameter output